JonathanCoco/ninjagold/apps/ninjagoldapp/views.py

```diff
@@ -0,0 +1,64 @@
+from django.shortcuts import render, redirect, HttpResponse
+import random
+import time
+
+# Create your views here.
+def index(request):
+
+    if ('your_gold' in request.session) == False:
+        request.session['your_gold'] = 0
+        request.session['your_activities'] = []
+
+    return render(request, 'ninjagoldapp/index.html')
+
+def process_money(request, building):
+
+    CurrMoney = int(request.session['your_gold'])
+
+    house = building
+    color = 'green'
+    activity_item = {}
+
+    money = 0
+    activity = ''
+
+    if house == "farm":
+        money = random.randrange(10, 20)
+    elif house == "cave":
+        money = random.randrange(5, 10)
+    elif house == "house":
+        money = random.randrange(2, 5)
+    elif house == "casino":
+        money =  random.randrange(-50, 50)
+
+    if money > 0:
+        activity = 'Earned {} from the {} on ({}).'.format(money, house, time.strftime("%c"))
+    else:
+        activity = 'Lost {} from the {} on ({}). Ouch!!!'.format(money, house, time.strftime("%c"))
+
+    CurrMoney = CurrMoney + money
+
```

Comment:
There's a few unnecessary variables here like 'house' and CurrMoney.  Every spot that you refer to 'house' you can just use building since they all have access to it.  With CurrMoney you can delete line 16 and then on line 39 just use 'request.session['your_gold'] += money'